### PR TITLE
Add provider & accrediting_provider views course

### PR DESCRIPTION
**Pull request by CatalinVoineag**
### Context

The course show page for find and publish(preview page) has too much
content. This commit tries to reduce this content by putting the
provider of the course and the accrediting provider of the course into
their own views.

For this to happen we also needed to change the logic of the
redirects/back links when adding a provider through the course show
page. You can do this when you have a draft course but without a
provider.

### Changes proposed in this pull request

New Summary list with new order:
- Fee or salary
- Course fee
- Course length
- Age range
- Qualification
- Provider
- Accredited by
- Date course starts
- Visa sponsorship

New view for provider (find and publish/preview)
New view for accredited provider (find and publish/preview)

Remove the provider and accredited provider info from the course show for publish and find.

### Guidance to review

Go on publish preview course page and check the links to the provider and accredited provider in the summary list.

Do the same for find

The provider and accredited provider should not be on course show page anymore

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [ ] Inform data insights team due to database changes

https://github.com/DFE-Digital/publish-teacher-training/assets/11318084/f82183f2-b2db-4007-a395-eb2e96ee4e84

### CODE REVIEW

**Comment by CatalinVoineag** [comment]
[Simplify course pages 4/6 - Move "About the training provider", "About the accredited provider", and "Contact this training provider" to separate pages and update course summary](https://trello.com/c/G1c5g8vM/1763-simplify-course-pages-4-6-move-about-the-training-provider-about-the-accredited-provider-and-contact-this-training-provider-to-s)

File: app/components/find/courses/about_schools_component/view.rb
Comment (CatalinVoineag) [inline app/components/find/courses/about_schools_component/view.rb:43]: Discovered a small edge case where you could bypass the old check. Using this `preview` should fix it

File: app/controllers/concerns/goto_preview.rb
Comment (CatalinVoineag) [inline app/controllers/concerns/goto_preview.rb:11]: We need to go back to the new provider page in some cases where there's no provider setup. Previously it would go to the preview page of the course but with moving the provider into its own view, we need to go back to that view not the preview

**Comment by kelliedesigner** [comment]
We're using some bespoke and inline css here where we could be using design system components, particularly to set vertical spacing. Will add detail in comments in context.

[Layout](https://design-system.service.gov.uk/styles/layout/)
[Summary list - no boards](https://design-system.service.gov.uk/components/summary-list#removing-the-borders)

File: app/views/publish/courses/accredited_provider/show.html.erb
Comment (dcyoung-dev) [inline app/views/publish/courses/accredited_provider/show.html.erb:4]: Should this be the Accredited Provider's name here?

File: app/views/publish/courses/preview.html.erb
Comment (dcyoung-dev) [inline app/views/publish/courses/preview.html.erb:18]: This seems like an unrelated removal?

File: spec/components/find/courses/about_schools_component/view_spec.rb
Comment (dcyoung-dev) [inline spec/components/find/courses/about_schools_component/view_spec.rb:16]: Could we set `preview: false` in the component's initialiser?

**Review by elceebee — APPROVED** [review]
Looks good! Approving with the assumption that you'll fix the styling issues flagged by Kellie before merging. Nice work getting up-to-speed with this codebase so quickly.

File: spec/helpers/find/goto_preview_helper_spec.rb
Comment (dcyoung-dev) [inline spec/helpers/find/goto_preview_helper_spec.rb:85]:
```suggestion
        it 'returns the value "true"' do
```
A bit of rogue copy and pasta

File: spec/helpers/find/goto_preview_helper_spec.rb
Comment (dcyoung-dev) [inline spec/helpers/find/goto_preview_helper_spec.rb:121]:
```suggestion
        let(:params) { { param_form_key: { goto_provider: 'false' } } }
```

Is it worth adding tests for this scenario?

File: app/components/find/courses/about_schools_component/view.rb
Comment (dcyoung-dev) [inline app/components/find/courses/about_schools_component/view.rb:17]:
```suggestion
        def initialize(course, preview: false)
```
This may stop make older uses of this Component from having to be updated

**Review by dcyoung-dev — COMMENTED** [review]
I've only looked at the code for this PR 👌
Don't let any of my comments be a blocker

File: app/controllers/find/courses/accrediting_providers_controller.rb
Comment (inulty-dfe) [inline app/controllers/find/courses/accrediting_providers_controller.rb:15]: Is there a potential double render error here?

File: app/controllers/find/courses/providers_controller.rb
Comment (inulty-dfe) [inline app/controllers/find/courses/providers_controller.rb:15]: Potential double render error?

File: app/components/find/courses/about_schools_component/view.rb
Comment (inulty-dfe) [inline app/components/find/courses/about_schools_component/view.rb:17]: What do you think of making this preview option default to false?

**Review by inulty-dfe — COMMENTED** [review]
Looking good, few questions

File: app/views/publish/courses/preview.html.erb
Comment (CatalinVoineag) [reply app/views/publish/courses/preview.html.erb:18]: No, this is meant to be removed. It's not very clear in the ticket but on the design in Lucid this is removed.

On the ticket it says it's done when `The course summary is updated, with new information added with links to the new views, other information removed, and the order of the rows updated`

And the lucid board indicates that this should be removed https://lucid.app/lucidspark/971f26f4-cc5e-4993-8f70-efc85b3f0772/edit?invitationId=inv_f64ad81b-8b03-472c-898d-e6c398214ede&page=0_0#

Screenshot from 2024-07-08 16-45-29

File: app/views/publish/courses/accredited_provider/show.html.erb
Comment (CatalinVoineag) [reply app/views/publish/courses/accredited_provider/show.html.erb:4]: Good catch

File: app/controllers/find/courses/providers_controller.rb
Comment (CatalinVoineag) [reply app/controllers/find/courses/providers_controller.rb:15]: Possibly, I'll change it. Good catch

File: app/controllers/find/courses/accrediting_providers_controller.rb
Comment (CatalinVoineag) [reply app/controllers/find/courses/accrediting_providers_controller.rb:15]: Possibly, I'll change it. Good catch

File: app/components/find/courses/summary_component/view.rb
Comment (CatalinVoineag) [reply app/components/find/courses/summary_component/view.rb:79]: Good shout

File: app/components/find/courses/qualifications_summary_component/view.html.erb
Comment (CatalinVoineag) [inline app/components/find/courses/qualifications_summary_component/view.html.erb:1]: This needs to be removed to fit nicely in the summary view component

**Comment by kelliedesigner** [comment]
## About provider

### Make whole content area two thirds width

Currently only the 'About [provider]' section is nested in the two thirds width.

Move `gov.uk-grid-row` and `govuk-grid-column-two-thirds` directly after `govuk-main-wrapper` to wrap the full content area.

| Example of the two thirds column area |
| --- |
|  Screenshot 2024-07-09 at 10 43 38 |

### Convert contact details to [summary list with no borders](https://design-system.service.gov.uk/components/summary-list/)

The contact details can be converted to use the summary list component, which will add more space to the rows.

### Additional content changes, not in the original design card

- Move website to be the first row
- Update label of `Website` to `School website`
- Update label of `Email` to `Email address`

## Course page

### Style secondary info under "Fee - no salary".

| Example |
| -- |
| Screenshot 2024-07-09 at 10 32 37 |

There isn't a set class for this. `govuk-hint` uses the correct colour but the font size is larger.

Therefore you could potentially do it in one of these two ways:
- use the [font override class](https://design-system.service.gov.uk/styles/font-override-classes/) of `https://design-system.service.gov.uk/styles/font-override-classes/`  with `govuk-hint`
- use `govuk-body-s` with the [colour](https://design-system.service.gov.uk/styles/colour/) ` $govuk-secondary-text-colour` or `#505a5f`

**Comment by kelliedesigner** [comment]
Looks good to me!

**Review by kelliedesigner — APPROVED** [review]

**Review by inulty-dfe — APPROVED** [review]
GTG

**Pull request merged**
